apps/form-management-app/src/app/state/form/form.slice.ts

```diff
@@ -1,64 +1,286 @@
+            ...result,
+            oneFormPerApplicant: typeof result.oneFormPerApplicant !== 'boolean' || result.oneFormPerApplicant,
+            urn: `${CONFIGURATION_SERVICE_ID}:v2:/configuration/form-service/${result.id}`,
+          })) || [],
+      };
+    } catch (err) {
+      if (axios.isAxiosError(err)) {
+        return rejectWithValue({
+          status: err.response?.status,
+          message: err.response?.data?.errorMessage || err.message,
+        });
+      } else {
+        throw err;
       }
-    );
-
-    return data.results;
-  } catch (err) {
-    if (axios.isAxiosError(err)) {
-      return rejectWithValue({
-        status: err.response?.status,
-        message: err.response?.data?.errorMessage || err.message,
-      });
-    } else {
-      throw err;
     }
   }
+);
+
+export const getFormSubmissions = createAsyncThunk(
+  'form/get-submissions',
+  async (
+    params: {
+      definitionId: string;
+      after?: string;
+      criteria?: FormSubmissionCriteria;
+    },
+    { getState, rejectWithValue }
+  ) => {
+    try {
+      const { config } = getState() as AppState;
+      const formServiceUrl = config.directory[FORM_SERVICE_ID];
+      const accessToken = await getAccessToken();
+
+      const { data } = await axios.get<PagedResults<FormSubmission>>(
+        new URL('/form/v1/submissions', formServiceUrl).href,
+        {
+          headers: { Authorization: `Bearer ${accessToken}` },
+          params: {
+            top: 20,
+            after: params.after,
+            criteria: JSON.stringify({
+              ...params.criteria,
+              definitionIdEquals: params.definitionId,
+            }),
+          },
+        }
+      );
+
+      return data;
+    } catch (err) {
+      if (axios.isAxiosError(err)) {
+        return rejectWithValue({
+          status: err.response?.status,
+          message: err.response?.data?.errorMessage || err.message,
+        });
+      } else {
+        throw err;
+      }
+    }
+  }
+);
+
+export const getFormSubmission = createAsyncThunk(
+  'form/get-submission',
+  async (submissionId: string, { getState, rejectWithValue }) => {
+    try {
+      const { config } = getState() as AppState;
+      const formServiceUrl = config.directory[FORM_SERVICE_ID];
+      const accessToken = await getAccessToken();
+
+      const { data } = await axios.get<FormSubmission>(
+        new URL(`/form/v1/submissions/${submissionId}`, formServiceUrl).href,
+        {
+          headers: { Authorization: `Bearer ${accessToken}` },
+        }
+      );
+
+      return data;
+    } catch (err) {
+      if (axios.isAxiosError(err)) {
+        return rejectWithValue({
+          status: err.response?.status,
+          message: err.response?.data?.errorMessage || err.message,
+        });
+      } else {
+        throw err;
+      }
+    }
+  }
+);
+
+export const selectDefinition = createAsyncThunk('form/select-definition', (definitionId: string, { dispatch }) => {
```

Comment:
these 2 actions look like they are used for form preview container and the data for selecting the formId or definition id should be managed within the form preview container state and not in the slice. so i suggest moving this logic to the container level where definitionId or submission Id is selected by the components and the container uses these thunks to load the data and return it to the container directly